src/did-provider.ts

```diff
@@ -0,0 +1,58 @@
+import { createJWS } from 'did-jwt'
+import {
+  HandlerMethods,
+  RequestHandler,
+  RPCConnection,
+  RPCError,
+  RPCRequest,
+  RPCResponse,
+  createHandler,
+} from 'rpc-utils'
+
+import IdentityWallet from './identity-wallet'
+
+type Origin = string | null | undefined
+
+type Context = {
+  wallet: IdentityWallet
+  origin: Origin
+}
+
+interface CreateJWSParams {
+  payload: Record<string, any>
+  protected?: Record<string, any>
+  pubKeyId?: string
+}
+
+const methods: HandlerMethods<Context> = {
+  did_authenticate: async ({ wallet, origin }) => {
+    await wallet.authenticate([], {}, origin)
+    return { did: wallet.DID }
+  },
+  did_createJWS: async ({ wallet, origin }, params: CreateJWSParams) => {
+    if (!(await wallet.isAuthenticated([], origin))) {
+      throw new RPCError(0, 'Authentication required')
+    }
+    const signer = wallet.getRootSigner(params.pubKeyId)
+    const jws = await createJWS(params.payload, signer, params.protected)
```

Comment:
We need to use `fast-json-stable-stringify` on the payload and the protected header hear so that it gets canonicalized properly!

Reply:
@oed changed in last commit.